x-pack/plugin/autoscaling/src/main/java/org/elasticsearch/xpack/autoscaling/storage/ReactiveStorageDeciderService.java

```diff
@@ -915,6 +965,13 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
             builder.field("assigned", assigned);
             builder.field("assigned_shards", assignedShardIds.stream().limit(MAX_AMOUNT_OF_SHARDS).toList());
             builder.field("assigned_shards_count", assignedShardIds.size());
+            AbstractAllocationDecision.nodeDecisionsToXContent(
+                unassignedAllocationResults,
+                builder,
+                params,
+                "unassigned_allocation_results"
+            );
+            AbstractAllocationDecision.nodeDecisionsToXContent(assignedAllocationResults, builder, params, "assigned_allocation_results");
```

Comment:
For the assigned shards, it is relevant to know both why it cannot remain on the current node and why it cannot be allocated to any node in the tier.